File: src/day43_Abstraction/Employee/Employee.java
package day43_Abstraction.Employee;

public abstract class Employee extends Person{

    private final int id;
    private String jobTitle;
    private double salary;

    protected Employee(final String name, final int age, final char gender, final int id, final String jobTitle, final double salary) {
        super(name, age, gender);
        this.id = id;
        this.jobTitle = jobTitle;
        this.salary = salary;
    }

    public int getId() {
        return this.id;
    }

    public String getJobTitle() {
        return this.jobTitle;
    }

    public void setJobTitle(final String jobTitle) {
        this.jobTitle = jobTitle;
    }

    public double getSalary() {
        return this.salary;
    }

    public void setSalary(final double salary) {
        this.salary = salary;
    }

    public abstract void work();

    public String toString() {
        return "Employee{" +
                "name=" + getName() +
                ", age=" + getAge() +
                ", gender=" + getGender() +
                ", id=" + id +
                ", jobTitle='" + jobTitle + '\'' +
                ", salary=" + salary +
                '}';
    }

File: src/day37_Inheritance/scrumTask/Employee.kt
package day37_Inheritance.scrumTask

class Employee(name: String, age: Int, gender: Char, //Employee is A Person
               var jobTitle: String, var id: Int, var salary: Double, var companyName: String) : Person(name, age, gender) {
    fun work() {
        println(jobTitle + " " + name + "is working")
    }

    override fun toString(): String {
        return "Employee{" +
                "jobTitle='" + jobTitle + '\'' +
                ", id=" + id +
                ", salary=" + salary +
                ", companyName='" + companyName + '\'' +
                ", name='" + name + '\'' +
                ", age=" + age +
                ", gender=" + gender +
                '}'
    }
}